### PR TITLE
Test/add e2e tests

### PR DESCRIPTION
**Pull request by raphael-arce**

### CODE REVIEW

File: tests/lists-page.spec.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  // Clicks on the title
+  await pageWithRecurrentList.getByPlaceholder('Title').click();
+
+  // Changes title to Breakfast
+  await pageWithRecurrentList.getByPlaceholder('Title').fill('Breakfast');
+  await pageWithRecurrentList.getByPlaceholder('Title').press('Enter');
+
+  // Returns to recurrent lists to save
+  await pageWithRecurrentList.locator('.text-1xl').click();
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:22]: missing expect to check the list name actually got changed (e.g. check on the list page if the new name is visible)

File: tests/lists-page.spec.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  //locates the background when back in light theme
+  const backgroundLight2 = pageWithRecurrentList.getByText('Groceries0Recurrent');
+
+  //gets the color from the background when back in light theme
+  const backgroundColorLight2 = await backgroundLight2.evaluate((e) => {
+    return window.getComputedStyle(e).getPropertyValue('color');
+  });
+
+  //checks the theme is back to the same light color as before
+  await expect(backgroundColorLight).toEqual(backgroundColorLight2);
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:94]: while it works, i'd suggest to rather do something like this:

```typescript
  //locates the background when in initial (light) theme
  const html = pageWithRecurrentList.locator('html');

  //gets the color from the background in light theme
  const initialBackgroundColor = await html.evaluate((e) => {
    return window.getComputedStyle(e).getPropertyValue('background-color');
  });

  expect(initialBackgroundColor).toEqual('rgb(241, 244, 242)');

  //changes theme to dark
  await pageWithRecurrentList
    .locator('div')
    .filter({ hasText: /^Groceries0$/ })
    .getByRole('button')
    .first()
    .click();

  const backgroundColorDark = await html.evaluate((e) => {
    return window.getComputedStyle(e).getPropertyValue('background-color');
  });

  expect(backgroundColorDark).toEqual('rgb(15, 23, 41)');

  await pageWithRecurrentList
    .locator('div')
    .filter({ hasText: /^Groceries0$/ })
    .getByRole('button')
    .first()
    .click();

  //gets the color from the background in light theme
  const backgroundColorLight = await html.evaluate((e) => {
    return window.getComputedStyle(e).getPropertyValue('background-color');
  });

  expect(backgroundColorLight).toEqual('rgb(241, 244, 242)');
```

File: tests/lists-page.spec.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  //Clicks on button to change to alphabetic sorting
+  await pageWithTwoRecurrentLists.locator('.relative > button').click();
+  await pageWithTwoRecurrentLists.getByRole('button', { name: 'A-Z' }).click();
+  await pageWithTwoRecurrentLists
+    .locator('div')
+    .filter({ hasText: /^Recurrent listsDateA-Z$/ })
+    .getByRole('button')
+    .nth(1)
+    .click();
+
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:107]: needs an expect before changing the sorting to prove that something actually changed

File: tests/lists-page.spec.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  //Clicks on button to change to alphabetic order
+  await pageWithTwoRecurrentLists.locator('.relative > button').click();
+  await pageWithTwoRecurrentLists.getByRole('button', { name: 'A-Z' }).click();
+  await pageWithTwoRecurrentLists
+    .locator('div')
+    .filter({ hasText: /^Recurrent listsDateA-Z$/ })
+    .getByRole('button')
+    .nth(1)
+    .click();
+
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:132]: as this is the default sorting, nothing is actually changing here. I think it's better to change it to a different sorting before changing it back to A-Z

File: tests/lists-page.spec.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  //Checks if first item on the list page is Breaky (ahead alphabetically of Lunch)
+  await expect(pageWithTwoRecurrentLists.getByRole('listitem').nth(0)).toHaveText(['BreakyStrawberries']);
+});
+
+test('should search for a list called Breaky', async ({ pageWithTwoRecurrentLists }) => {
+  await pageWithTwoRecurrentLists.locator('button:nth-child(3)').first().click();
+  await pageWithTwoRecurrentLists.getByPlaceholder('Looking for a list or an item').click();
+  await pageWithTwoRecurrentLists.getByPlaceholder('Looking for a list or an item').fill('Breaky');
+  await expect(pageWithTwoRecurrentLists.getByRole('listitem')).toHaveText(['BreakyStrawberries']);
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:141]: test would be more robust if we would start with multiple lists (e.g. 3, and maybe 2 of them starting with 'Breaky') to show that the search also works with multiple results

File: tests/lists-page.spec.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+});
+
+test('should search for an item called Strawberries', async ({ pageWithTwoRecurrentLists }) => {
+  await pageWithTwoRecurrentLists.locator('button:nth-child(3)').first().click();
+  await pageWithTwoRecurrentLists.getByPlaceholder('Looking for a list or an item').click();
+  await pageWithTwoRecurrentLists.getByPlaceholder('Looking for a list or an item').fill('Strawberries');
+  await expect(pageWithTwoRecurrentLists.getByRole('listitem')).toHaveText(['BreakyStrawberries']);
```
Comment (raphael-arce) [inline tests/lists-page.spec.js:148]: test would be more robust if we would start with multiple lists (e.g. 3, and maybe 2 of them having strawberries as item) to show that the search also works with multiple results

**Review by raphael-arce — COMMENTED** [review]
@samiletuns here are my suggestions 👍

File: tests/lists-page.spec.js (hunk 1 above)
Comment (samiletuns) [reply tests/lists-page.spec.js:22]: Added the check in https://github.com/GroceryFlux/GroceriesApp/pull/47/commits/57a7d5e9efa1e6cb471996c06c6df98b0e8edaa7

File: tests/lists-page.spec.js (hunk 2 above)
Comment (samiletuns) [reply tests/lists-page.spec.js]: Rewrote it in https://github.com/GroceryFlux/GroceriesApp/pull/47/commits/57a7d5e9efa1e6cb471996c06c6df98b0e8edaa7

File: tests/lists-page.spec.js (hunk 3 above)
Comment (samiletuns) [reply tests/lists-page.spec.js:107]: Right, this has been added in https://github.com/GroceryFlux/GroceriesApp/pull/47/commits/57a7d5e9efa1e6cb471996c06c6df98b0e8edaa7

File: tests/lists-page.spec.js (hunk 4 above)
Comment (samiletuns) [reply tests/lists-page.spec.js:148]: Added to check if it changes correctly to last edited and then back to A-Z in https://github.com/GroceryFlux/GroceriesApp/pull/47/commits/57a7d5e9efa1e6cb471996c06c6df98b0e8edaa7

File: tests/lists-page.spec.js (hunk 5 above)
Comment (samiletuns) [reply tests/lists-page.spec.js]: That's true ! Created two extra recurrent list (also addressing the next comment's issue) to have 3 lists, and in one case 2 similar names and in the other 2 lists with the same item 
https://github.com/GroceryFlux/GroceriesApp/pull/47/commits/57a7d5e9efa1e6cb471996c06c6df98b0e8edaa7

File: tests/lists-page.spec.js (hunk 6 above)
Comment (samiletuns) [reply tests/lists-page.spec.js]: See previous comment

**Pull request merged**